### DISCUSSION TITLE
Turbulence intensity in floris interface

### DISCUSSION THREAD

**Discussion by maapasan** [NatLabRockies/floris#654] [General]
Hello,

I would like to ask if there are plans to expand de input data "turbulence" in floris interface.

While wind speed or wind direction are defined using arrays (which allow the simultaneous simulation of several wind inputs), the turbulence intensity is defined as float.

I understand that turbulence intensity is established for all combinations of direction and speed.

However, the turbulence intensity values are dependent on wind speed (turbulence intensity decrease with wind speed). Do you have plans to modify this restriction?

Kind regards,

María

![4-Normal-turbulence-intensity-versus-wind-speed-at-the-hub-height-from-4-of-the-sites](https://github.com/NREL/floris/assets/125026723/707ac007-26c6-4666-9e80-04b871d4a4cc)

**Comment by Bartdoekemeijer** [comment]
Hi @maapasan. Indeed, FLORIS by default calculates a full grid of wind directions and wind speeds for a single TI value. However, you can include a unique TI for each wind direction/wind speed using the `time_series` functionality. See https://github.com/NREL/floris/pull/400.

Best,
Bart

**Reply by maapasan** [reply]
Hello @Bartdoekemeijer 
Thank you very much for your response.
I have already tried to use the time_series functionality, but I think that I am not doing it well.

I have tried to use 3 np.arrays (similar shape) for turbulence intensity, wind speed and wind direction. But I have obtained an error related with the type of turbulence intensity.
How should the time series functionality be used?

Thank you very much and kind regards,

Maria

![Captura](https://github.com/NREL/floris/assets/125026723/f75eeee0-a1d4-448a-80bb-a0970d0e5f1e)

**Reply by Bartdoekemeijer** [reply]
Ah, in that case I may be mistaken. You would need to write a separate call for each unique turbulence intensity value that you wish to simulate.

```
for ti in np.arange(0.01, 0.10, 0.01):
    fi.reinitialize(turbulence_intensity=ti)
    fi.calculate_wake()
    fi.get_turbine_powers()
    ...
```

**Reply by maapasan** [reply]
Hello @Bartdoekemeijer ,
In that case, do you have plans to extend that variable (or others) to array type to allow this?
Thank you!

María

**Reply by rafmudaf** [reply]
Hi @maapasan thanks for the question. I think this is a good suggestion for ongoing development of new features. Could you please describe this in an [issue](https://github.com/nrel/floris/issues)? Any context such as use case, how you'd like to specify the data, maybe a script that doesn't work yet but would demonstrate your idea, etc would be very helpful.
